core/include/drivers/nvmem.h

```diff
@@ -0,0 +1,188 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright (c) 2023, Microchip
+ */
+
+#ifndef __DRIVERS_NVMEM_H
+#define __DRIVERS_NVMEM_H
+
+#include <kernel/dt_driver.h>
+#include <tee_api_defines.h>
+#include <tee_api_types.h>
+#include <types_ext.h>
+
+struct nvmem_cell;
+
+/*
+ * struct nvmem_ops - Nvmem device driver operations
+ * @alloc_read_cell: Allocate @data in the heap and load @len bytes to from an
+ * nvmem cell
+ * @free_cell: Release resources allocated from nvmem_dt_get_func callback
+ * function
+ */
+struct nvmem_ops {
+	TEE_Result (*read_cell)(struct nvmem_cell *cell, uint8_t **data,
+				size_t *len);
+	void (*free_cell)(struct nvmem_cell *cell);
+};
+
+/*
+ * struct nvmem_cell - Description of an nvmem cell
+ * @offset: Cell byte offset in the NVMEM device
+ * @len: Cell byte size
+ * @ops: Nvmem device driver operation handlers
+ * @drv_data: Nvmem device driver private data
+ */
+struct nvmem_cell {
+	paddr_t offset;
+	size_t len;
+	const struct nvmem_ops *ops;
+	void *drv_data;
+};
+
+/*
+ * nvmem_dt_get_func - Typedef of function to get an nvmem reference from a DT
+ * node
+ *
+ * @a: Reference to phandle arguments
+ * @data: Pointer to data given at nvmem_dt_get_func() call
+ * @res: Output result code of the operation:
+ *	TEE_SUCCESS in case of success
+ *	TEE_ERROR_DEFER_DRIVER_INIT if clock is not initialized
+ *	Any TEE_Result compliant code in case of error.
+ */
+typedef
+struct nvmem_cell *(*nvmem_dt_get_func)(struct dt_pargs *a,
+					void *data, TEE_Result *res);
```

Comment:
You need to update to the new prototype, see https://github.com/OP-TEE/optee_os/pull/6098.
Here, somthing like:
```c
/*
 * nvmem_dt_get_func - Typedef of handlers to get an nvmem cell from a npode
 * @a: Reference to phandle arguments
 * @data: Pointer to data given at nvmem_dt_get_func() call
 * @out_cell: Output reference to cell instance upon success
 *
 * Return TEE_SUCCESS in case of success.
 * Return TEE_ERROR_DEFER_DRIVER_INIT if clock is not initialized
 * Return another TEE_Result compliant code otherwise.
 */
typedef TEE_Result (*nvmem_dt_get_func)(struct dt_pargs *a,
					void *data, struct nvmem_cell **out_cel);